extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsBuildTimeConfig.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.kafka.streams.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "kafka-streams", phase = ConfigPhase.BUILD_TIME)
```

Comment:
Interesting that it is set to `BUILD_TIME`. Do other extensions handle it like that, too?

Reply:
Only two extensions are handling it like that

extensions/undertow/deployment/.../ServletConfig.java:
```
@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
```

extensions/security/deployment/.../SecurityConfig.java:
```
@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
```

So as far as I understand, if the config is used at `BUILD_TIME` it has to be in the `deployment` part of the extension, and not in the `runtime` one.

Let me refactor this.

Reply:
> Only two extensions are handling it like that

Yes, so eventually the question is: should health check being enabled or not a build or runtime decision? Personally I feel it should be the latter, but if other extensions fix it at build time, we should align. // CC @gsmet

Reply:
Hmmm. My bad. I thought your question was about "only BUILD_TIME". I misunderstood.

Other extensions with health checks manage this at both with BUILD_AND_RUN_TIME.

Let me fix that. You're right.

Reply:
Adding health check is a build decision: why do you ever wanted to disabled an health heck in one env and not in the other, Just don't use it !

Adding it at build time allow to totaly remove all health check related stuff from runtime, this is the decision taken during the implementation of the infrastructure of the extension health check (and it is done like this in Agroal).

If you have a `MyExtensionBuildTimeConfig` object that provides only build time config, you can put it inside your deployement module so it will not end up being packaged inside your jar if you are using JVM mode. Few extensions does this, not all ...

Reply:
Ok, cool, let's do it as other extensions then and fix it at build time.